code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION!="Adults with Incapacity Act" & reas1=="Code 9", "Other code 9 reasons(not AWI)",reas2))
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Awaiting availability of transport","Transport",reas2))
+
+
+#write_sav(datafile3,paste0(filepath,"main census file Tabs 1 3 6.sav")) # save out file
+
+### 6. Get reason / age_grp breakdown for Scotland ----
+
+datafile4<-datafile3
+datafile4$level<-"1"
+
+datafile4$areaname<-"Scotland"
+
+Scotlandbymainreasongrouping <- datafile4 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas1) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(Scotlandbymainreasongrouping,paste0(filepath,"Scotland by main reason grouping.sav")) # save out file
+
+
+Scotlandbysubreasongrouping <- datafile4 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas2) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+
+
+
+#write_sav(Scotlandbysubreasongrouping,paste0(filepath,"Scotland by sub reason grouping.sav")) # save out file
+
+### 7. Breakdown for HBs (Level=2) ----
+datafile4$level<-"2"
+datafile4<-datafile4 %>% mutate(areaname=Healthboard)
```

Comment:
See "Style Guide", "Pipes", and "Code Needs Refactoring" sections in general feedback comments.

```suggestion
datafile %<>% mutate(level = "2", areaname = "Healthboard")
```